### PR TITLE
Bare excepts are deprecated, fix a warning in unittest

### PR DESCRIPTION
**Pull request by kraptor**
I've been getting this recently:

```
/home/kraptor/.choosenim/toolchains/nim-#devel/lib/pure/unittest.nim(553, 5) Warning: The bare except clause is deprecated; use `except CatchableError:` instead [BareExcept]
```

### CODE REVIEW

File: lib/pure/unittest.nim [hunk 1]
```diff
@@ -550,7 +550,7 @@ template test*(name, body) {.dirty.} =
         defer: testTeardownIMPL()
       body
 
-    except:
+    except CatchableError:
```
Comment (Menduist) [inline lib/pure/unittest.nim]:
```suggestion
    except CatchableError, Defect:
```
Unit test are one of the rare places where catching defect makes sense

Comment (ringabout) [reply lib/pure/unittest.nim]: Actually they are not completely equal, bare except can catch backend exceptions.

Comment (kraptor) [reply lib/pure/unittest.nim]: What to do then... do we just silence the warning?

Comment (arnetheduck) [reply lib/pure/unittest.nim]: > catching defect makes sense

might even make sense to `catch Exception` in such cases - ie the undefined behavior that follows doesn't really matter

> catch backend exceptions

only in c++ mode though, and not in v2 with goto exceptions.. ?

Comment (kraptor) [reply lib/pure/unittest.nim]: I updated it to catch `Exception` directly... and so be it. Let me know if we should update the code, but there are not many other alternatives.

**Comment by Araq** [comment]
Breaks tests/js/tunittest_error2.nim and seems to be a bad idea in general. I think in this context we really want to handle everything.

**Comment by zevv** [comment]
Yeah but this leaves my unit tests spewing out thousands of warnings cluttering the log:

```
/home/ico/external/Nim/lib/pure/unittest.nim(766, 5) Warning: The bare except clause is deprecated; use `except CatchableError:` instead [BareExcept]
```

**Comment by ringabout** [comment]
See also https://github.com/nim-lang/Nim/pull/21271

**Comment by ringabout** [comment]
Improved by https://github.com/nim-lang/Nim/pull/24821 When we have `raise {.foreign.}` we might undeprecate bare except warnings

**Pull request closed without merging**